### PR TITLE
feat: update email template

### PR DESCRIPTION
**Pull request by GreyyDaze**
### Description:

Develop an API endpoint that allows super admins to update existing email templates in the system.

### What does the PR do?

This PR implements a PATCH endpoint at `/api/v1/email/update-templates/:name` for updating email templates. The endpoint performs the following functions:
- Allows super admins to update existing email templates.
- Includes data validation to verify the correctness of the update.
- Provides error handling for scenarios such as authentication failures, template not found, and validation issues.

### How should this be manually tested?

To manually test the update email template endpoint, follow these steps:

1. **Open Postman.**

2. **Set the request type to PATCH.**

3. **Enter the API URL:**
   ```
   http://localhost:3000/api/v1/email/update-templates/{name}
   ```
   Replace `{name}` with the name of the email template you want to update.

4. **Authentication Testing:**
   - Ensure you have a valid super admin token.
   - Add the Bearer token to the Authorization header:
     ```
     Authorization: Bearer YOUR_VALID_TOKEN
     ```

5. **Successful Response:**
   - Send a PATCH request with a valid name and content in the request body:
     ```json
     {
       "templateName": "New Template Name",
       "template": "Updated email template"
     }
     ```
   - Check the response:
     - Status code should be 200 OK.
     - Response body should include the updated template details:
       ```json
       {
         "status_code": 200,
         "message": "Template updated successfully",
         "template": {
           "name": "New Template Name",
           "content": "updated html content"
         }
       }
       ```

6. **Authentication Error:**
   - Send a PATCH request without the Bearer token or with an invalid token.
   - Check the response:
     - Status code should be 401 Unauthorized.
     - Response body should include the message:
       ```json
       {
         "message": "User is currently unauthorized, kindly authenticate to continue.",
         "status_code": 401
       }
       ```

7. **Invalid Template Name Error:**
   - Send a PATCH request to `/api/v1/email/update-templates/{invalid_name}` with a non-existent or invalid name.
   - Check the response:
     - Status code should be 404 Not Found.
     - Response body should include the message:
       ```json
       {
         "message": "Template not found.",
         "status_code": 404
       }
       ```

8. **Insufficient Permissions Error:**
   - Send a PATCH request to `/api/v1/email/update-templates/{name}` with a role other than super admin.
   - Check the response:
     - Status code should be 403 Forbidden.
     - Response body should include the message:
       ```json
       {
         "message": "You don't have permission to perform this action.",
         "status_code": 403
       }
       ```

9. **Validation Errors:**
   - Send a PATCH request with missing fields:
     ```json
     {
     }
     ```
   - Check the response:
     - Status code should be 400 Bad Request.
     - Response body should include the message:
       ```json
       {
         "message": [
           "templateName should not be empty",
           "template should not be empty",
           "template must be a string"
         ],
         "error": "Bad Request",
         "status_code": 400
       }
       ```

   - Send a PATCH request with invalid HTML format in the content field:
     ```json
     {
       "templateName": "Valid Name",
       "template": "<html><body>Invalid HTML"
     }
     ```
   - Check the response:
     - Status code should be 400 Bad Request.
     - Response body should include the message:
       ```json
       {
         "message": "Invalid HTML format.",
         "status_code": 400
       }
       ```

### Tasks

- [X] Implemented the PATCH endpoint `/api/v1/email/update-templates/:name` for updating email templates.
- [X] Integrated error handling for authentication issues, invalid or missing fields, and validation errors.
- [X] Added validation to ensure correct data format and content for email template updates.
- [X] Created unit tests to verify successful updates, handle authentication errors, and check for validation issues.
- [X] Updated API documentation to reflect the new endpoint and its usage.

### Output
success
template error

### CODE REVIEW

File: src/modules/email/email.controller.ts [hunk 1]
```diff
@@ -1,35 +1,63 @@
...
-import { skipAuth } from '../../helpers/skipAuth';
-import { SendEmailDto, createTemplateDto, getTemplateDto } from './dto/email.dto';
+import { SendEmailDto, UpdateTemplateDto, createTemplateDto, getTemplateDto } from './dto/email.dto';
 import { skip } from 'node:test';
-import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { ApiBearerAuth, ApiOperation, ApiParam, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { SuperAdminGuard } from '../../guards/super-admin.guard';
 
-@ApiTags('Emails')
+@ApiTags('email')
```
Comment (CFCIfe) [inline src/modules/email/email.controller.ts:9]: Api Tags should be uppercase. and can be left as Emails

**Review by CFCIfe — CHANGES_REQUESTED** [review]
Fix the issue with the API Tags. The code looks good. Re-request a review after

File: src/modules/email/email.controller.ts (hunk 1 above)
Comment (GreyyDaze) [reply src/modules/email/email.controller.ts]: fix

File: package.json [hunk 2]
```diff
@@ -32,7 +32,6 @@
     "seed": "ts-node src/seed.ts"
   },
   "dependencies": {
-    "@css-inline/css-inline-linux-x64-gnu": "^0.14.1",
```
Comment (CFCIfe) [inline package.json:35]: Return this package

```
    "@css-inline/css-inline-linux-x64-gnu": "^0.14.1",
```

File: src/modules/email/dto/email-response.dto.ts
Comment (CFCIfe) [inline src/modules/email/dto/email-response.dto.ts:1]: Split this into multiple DTOs

Comment (CFCIfe) [inline src/modules/email/dto/email-response.dto.ts:1]: Split this into multiple DTOs

**Review by CFCIfe — COMMENTED** [review]
Review comments

File: src/modules/email/dto/email-response.dto.ts
Comment (GreyyDaze) [reply src/modules/email/dto/email-response.dto.ts:1]: fix

Comment (GreyyDaze) [reply src/modules/email/dto/email-response.dto.ts:1]: fix

File: package.json (hunk 2 above)
Comment (GreyyDaze) [reply package.json]: fix

**Review by CFCIfe — DISMISSED** [review]

**Review by CFCIfe — APPROVED** [review]

**Pull request merged**